plugins/inputs/nsdp/nsdp.go

```diff
@@ -0,0 +1,121 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package nsdp
+
+import (
+	_ "embed"
+	"net"
+	"strconv"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/tdrn-org/go-nsdp"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+const pluginName = "nsdp"
+
+const defaultTimeout = config.Duration(2 * time.Second)
+
+type NSDP struct {
+	Target      string          `toml:"target"`
+	DeviceLimit uint            `toml:"device_limit"`
+	Timeout     config.Duration `toml:"timeout"`
+
+	Log telegraf.Logger `toml:"-"`
+}
+
+func defaultNSDP() *NSDP {
+	return &NSDP{
+		Target:  nsdp.IPv4BroadcastTarget,
+		Timeout: defaultTimeout,
+	}
+}
```

Comment:
Please set the `Target` member in the `Init()` function if the field is empty.

Reply:
Adapted via 8c321f127f426a5c1cbdc9876b56cf4409fafce2